projects/flimnap/main.cc

```diff
@@ -301,42 +317,61 @@ int main(void) {
         float steer_torque = 0.0f;
         (void)kistler_torque;
 #else // defined(FLIMNAP_ZERO_INPUT)
-        const float inertia_torque = -handlebar_model.torque(model_t::get_state_part(bicycle.full_state()));
+        const float inertia_torque = -handlebar_inertia.torque(model_t::get_state_part(bicycle.full_state()));
         float steer_torque = kistler_torque - inertia_torque;
 #endif // defined(FLIMNAP_ZERO_INPUT)
 
         // simulate bicycle
         bicycle.set_v(v);
-#if !defined(USE_BICYCLE_KINEMATIC_MODEL)
-        if (bicycle.v() < assistance_velocity_limit) {
-            const float interp = bicycle.v() / assistance_velocity_limit;
-            const model_t::input_t u = controller.control_calculate(bicycle.observer().state(), interp);
-            if (assistance_fade_counter < assistance_fade_period) {
-                ++assistance_fade_counter;
+        { // perform variant specific code for creating input/measurement
+#if defined(USE_BICYCLE_KINEMATIC_MODEL)
+            bicycle.update_dynamics(roll_torque, steer_torque, yaw_angle, steer_angle, rear_wheel_angle);
+#elif defined(USE_BICYCLE_AREND_MODEL)
+            // BicyleArend uses a different output/measurement vector definition
+            const float steer_rate = steer_rate_filter.output(
+                    (steer_angle - prev_steer_angle)/
+                    (static_cast<float>(ST2MS(dynamics_loop_period))/1000.0f));
+            prev_steer_angle = steer_angle;
+
+            model_t::input_t u = model_t::input_t::Zero();
+            model_t::set_input_element(u, model_t::input_index_t::roll_torque, roll_torque);
+            model_t::set_input_element(u, model_t::input_index_t::steer_torque, steer_torque);
+
+            model_t::measurement_t z = model_t::measurement_t::Zero();
+            bicycle.model().set_output_element(z, bicycle.model().output_index_t::steer_angle, steer_angle);
+            bicycle.model().set_output_element(z, bicycle.model().output_index_t::steer_rate, steer_rate);
+
+            bicycle.update_dynamics(u, z);
+#else
+            if (bicycle.v() < assistance_velocity_limit) {
```

Comment:
I understand the principle behind `fade` and how it is different from `interp`. I don't understand why both seem to be applied because I am *guessing* that in the control_calculate computation, `K0+ interp*Kd` affects the 'strength' of the assistance, just like `fade` does in `fade * model_t::get_input_element(u,  ...`. 

In the scenario where the rider has been riding above the `assistance_velocity_limit` for some time and suddenly breaks, the `assistance_fade_counter` needs some time to increase meaning no assistance will be applied while the bicycle is extremely unstable. 

What if we just use a function of `bicycle.v() / asistance_velocity_limit` to have a smooth transition instead of the on/off behaviour? That would make the assistance behaviour easier to predict and solve the unstability problem. 

![figure_1](https://user-images.githubusercontent.com/5444990/28675649-c6a6e73e-72e8-11e7-8a70-47bbbc838d45.png)